### ISSUE TITLE
Wrong colormap returned in `nx.draw_networkx_edges` 

### ISSUE THREAD

**Issue by igormorgado** [networkx/networkx#3744]
MWE:
```

import numpy as np
import matplotlib.pyplot as plt
import networkx as nx
import itertools

# Graph size
m = 5

# Create a random weights
weights = [ ((i, j), 2*np.random.random()-1) for i, j in itertools.product(np.arange(m), np.arange(m))]

# Create graph and set weights
G = nx.complete_graph(m)
nx.set_edge_attributes(G, dict(weights), 'weight')
edgelist, edge_color = zip(*nx.get_edge_attributes(G,'weight').items())

# Draw graph
fig, ax = plt.subplots(figsize=(5,5))
pos = nx.circular_layout(G)
nodes = nx.draw_networkx_nodes(G, pos, node_color='#aaaaaa', axes=ax)
edges = nx.draw_networkx_edges(G, pos, edgelist=edgelist, edge_color=edge_color, edge_cmap=plt.cm.coolwarm, axes=ax)
cb = plt.colorbar(edges, ax=ax)
plt.axis('off')
print(edges.cmap.name)
```

For some reason (bug perhaps), the `cmap`  returned by `nx.draw_networkx_edges` is set to `viridis` instead the one set in function call. Hence after the call to `draw_networkx_edges` I have set:

    edges.cmap = plt.cmap.coolwarm
    edges.set_clim(np.min(edge_color),np.max(edge_color))

And that works.

For sake of documentation NetworkX version is `2.4`. Stable in October, 2019.

**Comment by dschult** [comment]
Yup! That's a bug in #3696 that was fixed in #3698 
so it will be in the v2.5 release.  Thanks for the heads up and nice example code!!

**Comment by rossbar** [comment]
I'm going to close this since it seems it was fixed in #3698. Please feel free to reopen if you'd like to continue the discussion.
